delfin/drivers/ibm/storwize_svc/ssh_handler.py

```diff
@@ -0,0 +1,383 @@
+# Copyright 2020 The SODA Authors.
+# Copyright (c) 2016 Huawei Technologies Co., Ltd.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import time
+
+import paramiko
+import six
+from oslo_log import log as logging
+from delfin.drivers.utils.ssh_client import SSHPool
+from delfin.common import constants
+
+from delfin import exception
+
+LOG = logging.getLogger(__name__)
+
+
+class SSHHandler(object):
+
+    login_command = 'help'
+    systeminfo_command = 'lssystem'
+    poolinfo_command = 'lsmdiskgrp'
+    poolinfo_detail_command = 'lsmdiskgrp '
+    volumeinfo_command = 'lsvdisk'
+    volumeinfo_detail_command = 'lsvdisk -delim : '
+    eventlog_command = 'lseventlog -filtervalue "status=alert"'
+    eventlog_command_without_filter = 'lseventlog -monitoring yes'
+    alert_detail_command = 'lseventlog '
+    enclosure_command = 'lsenclosure -delim :'
+
+    TIME_PATTERN = '%Y%m%d%H%M%S'
+
+    SEVERITY_MAP = {"warning": "Warning",
+                    "informational": "Informational",
+                    "error": "Major"
+                    }
+
+    def __init__(self, **kwargs):
+        self.kwargs = kwargs
+        self.ssh_pool = SSHPool(**kwargs)
+
+    def set_storage_id(self, storage_id):
+        self.storage_id = storage_id
+
+    def login(self, context):
+        """Test SSH connection """
+        version = ''
+        try:
+            with self.ssh_pool.item() as ssh:
+                self.do_exec(SSHHandler.login_command, ssh)
+        except Exception as e:
+            msg = "Failed to login ibm storwize_svc %s" % (six.text_type(e))
+            raise exception.SSHException(msg)
+        return version
+
+    def unitstobytes(self, unit):
+        result = 0
+        if unit == 'TB':
+            result = 1024 * 1024 * 1024 * 1024
+        elif unit == 'GB':
+            result = 1024 * 1024 * 1024
+        elif unit == 'MB':
+            result = 1024 * 1024
+        elif unit == 'KB':
+            result = 1024
+        else:
+            result = 1
+        return int(result)
+
+    def do_exec(self, command_str, ssh):
+        """Execute command"""
+        re = None
```

Comment:
This is not used except the if block, so it may not need to be initialized here

Reply:
Has changed